specification/appendix-d-observability.md

```diff
@@ -0,0 +1,50 @@
+---
+id: appendix-d
+title: "Appendix D: Observability"
+description: Conventions for OpenFeature telemetry signals 
+sidebar_position: 5
+---
+
+# Appendix D: Observability
+
+This document describes conventions for extracting data from the OpenFeature SDK for use in telemetry signals.
+It primarily focuses on providing recommendations for mapping well-known fields in OpenFeature to [OpenTelemetry feature-flag log records](https://opentelemetry.io/docs/specs/semconv/feature-flags/feature-flags-logs/) and other semantic conventions.
+
+## Evaluations
+
+Flag evaluation telemetry comprises data resolved from the provider resolution (evaluation details and flag metadata) as well as metadata about the provider itself.
+This is particularly relevant to telemetry-related [hooks](./sections/04-hooks.md).
+
+### Evaluation Details
+
+The following describes how fields on the [evaluation details](types.md#evaluation-details) are mapped to feature flag log records:
+
+| Evaluation Details Field | Log Record Attribute                    | Notes                                                |
+| ------------------------ | --------------------------------------- | ---------------------------------------------------- |
+| `flag key`               | `feature_flag.key`                      | See: [flag key](./glossary.md#flag-key)              |
+| `error code`             | `error.type`                            | See: [error code](./types.md#error-code)             |
+| `variant`, `value`       | `feature_flag.variant`                  | See: [variant](./glossary.md#variant)                |
+| `error message`          | `feature_flag.evaluation.error.message` | An error message associated with a failed evaluation |
+| `reason`                 | `feature_flag.evaluation.reason`        | See: [reason](./types.md#reason)                     |
+
+> [!NOTE]  
+> If the provider in question doesn't supply a `variant`, the a string representation of the `value` is used.
```

Comment:
The OTel spec allows for `value` to be "any" type.

Reply:
I have removed "string".

Reply:
I've removed this whole note for now (see https://github.com/open-feature/spec/pull/287#discussion_r1887371546)